File: frontend-playstore/PusulaService/app/src/main/java/com/pusula/service/data/model/ProposalModels.kt
package com.pusula.service.data.model

data class ProposalDTO(
    val id: Long? = null,
    val companyId: Long? = null,
    val customerId: Long? = null,
    val customerName: String? = null,
    val preparedById: Long? = null,
    val preparedByName: String? = null,
    val status: String? = null,
    val validUntil: String? = null,
    val note: String? = null,
    val title: String? = null,
    val taxRate: Double? = null,
    val discount: Double? = null,
    val subtotal: Double? = null,
    val taxAmount: Double? = null,
    val totalPrice: Double? = null,
    val items: List<ProposalItemDTO> = emptyList()
)

data class ProposalItemDTO(
    val id: Long? = null,
    val description: String,
    val quantity: Int,
    val unitCost: Double? = null,
    val unitPrice: Double,
    val totalPrice: Double? = null
)

File: frontend-appstore/PusulaService/Models/ProposalModels.swift
import Foundation

struct ProposalDTO: Codable, Identifiable {
    var id: Int?
    let companyId: Int?
    let customerId: Int?
    let customerName: String?
    let preparedById: Int?
    let preparedByName: String?
    let status: String?
    let validUntil: String?
    let note: String?
    let title: String?
    let taxRate: Double?
    let discount: Double?
    let subtotal: Double?
    let taxAmount: Double?
    let totalPrice: Double?
    var items: [ProposalItemDTO]?
}

struct ProposalItemDTO: Codable, Identifiable {
    var id: Int?
    let description: String
    var quantity: Int
    let unitCost: Double?
    var unitPrice: Double
    var totalPrice: Double?
}
